sky/policy.py

```diff
@@ -1,154 +1,82 @@
-"""Customize policy by users."""
-import copy
+"""Interface for admin-defined policy for user requests."""
+import abc
 import dataclasses
-import importlib
-import os
-import tempfile
 import typing
-from typing import Any, Callable, Dict, Optional
+from typing import Optional
 
-from sky import dag as dag_lib
-from sky import sky_logging
-from sky import skypilot_config
-from sky.utils import common_utils
-from sky.utils import ux_utils
+if typing.TYPE_CHECKING:
+    import sky
 
-logger = sky_logging.init_logger(__name__)
 
-if typing.TYPE_CHECKING:
-    from sky import task as task_lib
+@dataclasses.dataclass
+class OperationArgs:
+    cluster_name: Optional[str]
+    cluster_exists: bool
+    idle_minutes_to_autostop: Optional[int]
+    down: bool
+    dryrun: bool
 
 
 @dataclasses.dataclass
-class UserTask:
-    task: 'task_lib.Task'
-    skypilot_config: Dict[str, Any]
+class UserRequest:
+    """User request to the policy.
+
+    Args:
+        task: User specified task.
+        skypilot_config: Global skypilot config.
+        execution_args: Execution arguments. It can be None for jobs and
+            services.
+    """
+    task: 'sky.Task'
+    skypilot_config: 'sky.NestedConfig'
+    operation_args: Optional['OperationArgs'] = None
 
 
 @dataclasses.dataclass
-class MutatedUserTask:
-    task: 'task_lib.Task'
-    skypilot_config: Dict[str, Any]
+class MutatedUserRequest:
+    task: 'sky.Task'
+    skypilot_config: 'sky.NestedConfig'
 
 
-class Policy:
-    """User-defined policy.
+# pylint: disable=line-too-long
+class AdminPolicy:
+    """Interface for admin-defined policy for user requests.
 
     A user-defined policy is a string to a python function that can be imported
     from the same environment where SkyPilot is running.
 
     It can be specified in the SkyPilot config file under the key 'policy', e.g.
```

Comment:
```suggestion
    A subclass can be registered in the SkyPilot config file under the key 'policy', e.g.
```
I'd move this para to below

Reply:
Rephrased it a bit to make the target audience more clear. How does it look?